refactor: update `math/base/assert/is-even` to follow latest project conventions

vivekmaurya001 opened the pull request:
Resolves None.

## Description

- refactoring addon.c to follow latest project conventions for  `math/base/assert/is-even` 
- updating benchmark calculations

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
refactor: update math/base/assert/is-even to follow latest project conventions

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into math/is-even

vivekmaurya001 committed:
refactor: update math/base/assert/is-even to follow latest project conventions

vivekmaurya001 committed:
Merge branch 'math/is-even' of https://github.com/vivekmaurya001/stdlib into math/is-even

vivekmaurya001 commented:
good afternoon, please give this a reveiw @kgryte!

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/benchmark.js:
```diff
@@ -21,6 +21,7 @@
 // MODULES //
 
 var bench = require( '@stdlib/bench' );
+var Float64Array = require( '@stdlib/array/float64' );
 var randu = require( '@stdlib/random/base/randu' );
 var round = require( '@stdlib/math/base/special/round' );
```
@vivekmaurya001 Instead of using a manual loop to generate random values, would you mind using `random/array/discrete-uniform`?

vivekmaurya001 replied:
done !

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/benchmark.native.js:
```diff
@@ -23,6 +23,7 @@
 var resolve = require( 'path' ).resolve;
 var bench = require( '@stdlib/bench' );
 var randu = require( '@stdlib/random/base/randu' );
+var Float64Array = require( '@stdlib/array/float64' );
 var round = require( '@stdlib/math/base/special/round' );
```
Same comment.

vivekmaurya001 replied:
done

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/c/native/benchmark.c:
```diff
@@ -92,15 +92,18 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( rand_double() * 200.0 ) - 100.0;
```
@vivekmaurya001 Would you mind rounding this values? Using the built-in C99 `round` function should be fine.

vivekmaurya001 replied:
yeah sorry i forgot to add it , now corrected !

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/src/addon.c:
```diff
@@ -29,60 +32,11 @@
-	return v;
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 1 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, x, argv, 0 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, stdlib_base_is_even( x ), out );
```
@vivekmaurya001 You should use `STDLIB_NAPI_CREATE_INT32` as done in the original bindings.

vivekmaurya001 replied:
okay ! , done

kgryte reviewed (changes requested):
Left an initial round of comments.

vivekmaurya001 committed:
refactor: update math/base/assert/is-even to follow latest project conventions

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
---

vivekmaurya001 committed:
refactor: update math/base/assert/is-even to follow latest project conventions

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into math/is-even

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/c/native/benchmark.c:
```diff
@@ -92,15 +92,18 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = round(( rand_double() * 200.0 ) - 100.0);
```
```suggestion
		x[ i ] = round( (rand_double()*200.0) - 100.0 );
```

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/src/addon.c:
```diff
@@ -17,9 +17,12 @@
+#include "stdlib/napi/create_int32.h"
+#include "stdlib/napi/export.h"
 #include <node_api.h>
 #include <stdint.h>
```
```suggestion
```

No longer used.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/src/addon.c:
```diff
@@ -17,9 +17,11 @@
+#include "stdlib/napi/argv_double.h"
+#include "stdlib/napi/create_int32.h"
+#include "stdlib/napi/export.h"
 #include <node_api.h>
```
```suggestion
#include <node_api.h>
#include <stdint.h>
```

Correction, it should be used below.

kgryte commented on lib/node_modules/@stdlib/math/base/assert/is-even/src/addon.c:
```diff
@@ -29,60 +31,11 @@
-	return v;
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 1 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, x, argv, 0 );
+	STDLIB_NAPI_CREATE_INT32( env, stdlib_base_is_even( x ), out );
```
```suggestion
	STDLIB_NAPI_CREATE_INT32( env, (int32_t)stdlib_base_is_even( x ), out );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
